analyses/tp53_nf1_score/01-apply-classifier.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Apply Machine Learning Classifiers to PDX RNAseq data
+# 
+# **Gregory Way, 2018**
+# 
+# In the following notebook, I apply three distinct classifiers to patient derived xenograft (PDX) RNAseq data (FPKM).
+# The first classifier detects Ras activation. For more details about the algorithm and results, refer to [Way et al. 2018](https://doi.org/10.1016/j.celrep.2018.03.046 "Machine Learning Detects Pan-cancer Ras Pathway Activation in The Cancer Genome Atlas"). I also include _TP53_ inactivation predictions. This classifier was previously applied in [Knijnenburg et al. 2018](https://doi.org/10.1016/j.celrep.2018.03.076 "Genomic and Molecular Landscape of DNA Damage Repair Deficiency across The Cancer Genome Atlas"). The third is a classifier that predicts _NF1_ inactivation. We previously applied this in [Way et al. 2017](https://doi.org/10.1186/s12864-017-3519-7 "A machine learning classifier trained on cancer transcriptomes detects NF1 inactivation signal in glioblastoma").
+# 
+# To apply other classifiers (targeting other genes of interest) refer to https://github.com/greenelab/pancancer.
+# 
+# Also note that we have implemented classifiers on a larger scale and made the models accessible to non-computational biologists (see [Project Cognoma](http://cognoma.org)).
+# 
+# ## Procedure
+# 
+# 1. Load RNAseq matrix (`.RDS`)
+#   * The matrix is in `sample` x `gene symbol` format (250 x 51,968)
+# 2. Process matrix
+#   * Take the z-score by gene
+# 3. Load classifier coefficients
+#   * For both _Ras_, _TP53_, and _NF1_ classifiers
+# 4. Apply each classifier to the input data
+#   * This also requires additional processing steps to the input data (subsetting to the respective classifier genes)
+#   * Also note that not all genes are present in the input RNAseq genes.
+# 5. Shuffle the gene expression data (by gene) and apply each classifier to random data
+# 
+# ### Important Caveat
+# 
+# Because not all of the classifier genes are found in the input dataset, the classifier probability is not calibrated correctly. The scores should be interpreted as continuous values representing relative gene alterations, and not as a pure probability estimate.
+# 
+# ## Output
+# 
+# The output of this notebook are the predicted scores for both classifiers across all samples for real data and shuffled data. This is in the form of a single text file with three columns (`sample_id`, `ras_score`, `tp53_score`, `nf1_score`,  `ras_shuffle`, `tp53_shuffle`, `nf1_shuffle`).
+
+# Modified by Krutika Gaonkar for OpenPBTA 2019
+
+
+import os
```

Comment:
In the PPTC analysis, this `.py` file was autogenerated from a `.ipynb` file with `nbconvert`. This is why there are some strangely formatted lines.

Consider reformatting manually with a linter, or by using [black](https://black.readthedocs.io/en/stable/). (I'd recommend using black - it is easy and consistent)